### PR TITLE
SB-25552 - UCI Inbound Deployment

### PR DESCRIPTION
**Pull request by shiva-rakshith**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Enhancement (additive changes to improve performance)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ansible/roles/stack-sunbird/tasks/inbound.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (keshavprasadms) [inline ansible/roles/stack-sunbird/tasks/inbound.yml:1]: We are no more supporting docker deployments. So this file is not required.

File: kubernetes/helm_charts/core/inbound/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          configMap:
+            name: {{ .Chart.Name }}-config
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.dockerhub }}/{{ .Values.repository }}:{{ .Values.image_tag }}"
+        imagePullPolicy: Always
+        env:
+        - name: JAVA_OPTIONS
+          value: {{ .Values.env.javaoptions | quote }}
+        envFrom:
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/inbound/templates/deployment.yaml:38]: What is the purpose of having configmap as env and also as a volume mount? Does the application require it to be present both as env variables and a locally mounted file in the pod?

File: kubernetes/helm_charts/core/inbound/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        {{- if .Values.healthcheck }}
+        livenessProbe:
+{{ toYaml .Values.livenessProbe | indent 10 }}
+        readinessProbe:
+{{ toYaml .Values.readinessProbe | indent 10 }}
+        {{- end }}
+        volumeMounts:
+          - name: {{ .Chart.Name }}-config
+            mountPath: /home/sunbird/inbound/config/application.properties
+            subPath: inbound.env
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/inbound/templates/deployment.yaml:54]: Why do we need a `subPath` here?

**Review by keshavprasadms — COMMENTED** [review]
@shiva-rakshith Please check my comments

File: ansible/roles/stack-sunbird/tasks/inbound.yml (hunk 1 above)
Comment (shiva-rakshith) [reply ansible/roles/stack-sunbird/tasks/inbound.yml]: Removed file.

File: kubernetes/helm_charts/core/inbound/templates/deployment.yaml (hunk 2 above)
Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/inbound/templates/deployment.yaml:38]: Verified with the team, the file is not needed locally mounted, made corresponding changes to code.

File: kubernetes/helm_charts/core/inbound/templates/deployment.yaml (hunk 3 above)
Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/inbound/templates/deployment.yaml]: Removed volumemounts as it is not necessary.

File: kubernetes/helm_charts/core/inbound/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  template:
+    metadata:
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+{{- if .Values.imagepullsecrets }}
+      imagePullSecrets:
+      - name: {{ .Values.imagepullsecrets }}
+{{- end }}
+      volumes:
```
Comment (keshavprasadms) [inline kubernetes/helm_charts/core/inbound/templates/deployment.yaml:27]: @shiva-rakshith Do we need this?

Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/inbound/templates/deployment.yaml:27]: As of now we don't have and secrets, So we don't need this, removing it.

Comment (keshavprasadms) [reply kubernetes/helm_charts/core/inbound/templates/deployment.yaml:23]: What I meant is,  you have removed volumeMounts but still have retained volumes. Please remove volumes section and add the imagePullSecrets back. We need the image secrets. Please refer to the correct line of comment in review.

Comment (shiva-rakshith) [reply kubernetes/helm_charts/core/inbound/templates/deployment.yaml:23]: Apologies for misunderstanding, i will make the correct changes.

**Review by keshavprasadms — APPROVED** [review]
Looks good!

**Pull request merged**